CMakeLists.txt

```diff
@@ -60,6 +60,8 @@ option(USE_ASAN "Enable Clang/GCC ASAN sanitizers." OFF)
 option(ENABLE_TESTCOVERAGE "Enable compilation with test coverage metric output" OFF)
 option(USE_INT64_TENSOR_SIZE "Use int64_t to represent the total number of elements in a tensor" OFF)
 option(BUILD_CYTHON_MODULES "Build cython modules." OFF)
+option(USE_SPLIT_ARCH_DLL   "Build mxnet with spilt different dll every cuda arch in windows." ON)
```

Comment:
It's only valid on Windows, so how about hiding the option via `cmake_dependent_option` on other systems.

`cmake_dependent_option(USE_SPLIT_ARCH_DLL "Build a separate DLL for each Cuda arch (Windows only)." ON "NOT MSVC" OFF)`